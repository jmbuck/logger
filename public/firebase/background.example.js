// Initialize Firebase
var config = {
    apiKey: "EXAMPLE",
    authDomain: "EXAMPLE",
    databaseURL: "EXAMPLE",
    projectId: "EXAMPLE",
    storageBucket: "EXAMPLE",
    messagingSenderId: "EXAMPLE"
};
firebase.initializeApp(config);

/**
 * initApp handles setting up the Firebase context and registering
 * callbacks for the auth status.
 *
 * The core initialization is in firebase.App - this is the glue class
 * which stores configuration. We provide an app name here to allow
 * distinguishing multiple app instances.
 *
 * This method also registers a listener with firebase.auth().onAuthStateChanged.
 * This listener is called when the user is signed in or out, and that
 * is where we update the UI.
 *
 * When signed in, we also authenticate to the Firebase Realtime Database.
 */
function initApp() {
    // Listen for auth state changes.
    firebase.auth().onAuthStateChanged(function(user) {
        console.log('User state change detected from the Background script of the Chrome Extension:', user);
    });
}

window.onload = function() {
    initApp();
};