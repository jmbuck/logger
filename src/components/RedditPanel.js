import React, {Component} from 'react'
import '../css/common.css'
import exitIcon from "../img/x.svg"
import DataNav from './DataNav'
import {auth, retrieveFirebaseUserRedditData} from "../rebase";

export class RedditPanel extends Component {

    constructor(props) {
        super(props);

        this.state = {
            data : []
        };
        this.handleDelete = this.handleDelete.bind(this);
    }

    retrieve = (user) => {
        if(user) {
            retrieveFirebaseUserRedditData(user.uid, (data) => {
                this.setState({ data : data})
            })
        }
    };

    componentWillMount() {
        if(auth.currentUser)
            this.retrieve(auth.currentUser);

        auth.onAuthStateChanged((user) => {
            this.retrieve(user);
        })
    }

    handleDelete = (index) => {
        let data = [...this.state.data];
        data.splice(index, 1);
        this.setState({data});
    };

    render() {
        return (
	        <div className="panel">
                <div className="panel-container">
                    <div className="panel-reddit">
	                    <DataNav {...this.props}/>
                        <div className="panel-center-content">
	                        <h1>Reddit</h1>
	                        <table border="1px solid black">
		                        <tr>
			                        <th>Subreddit</th>
			                        <th>Time Spent</th>
                                    <th>Delete</th>
		                        </tr>
                                {
                                    this.state.data.map((d, index) =>
                                        <tr>
                                            <td>{d.name}</td>
                                            <td>{d.time}</td>
                                            <td>
                                                <img src={exitIcon} onClick={() => this.handleDelete(index)} style={{"filter" : "invert(100%)", "width": "25px", "height" : "25px", "cursor" : "pointer"}}/>
                                            </td>
                                        </tr>)
                                }
	                        </table>
                        </div>
                    </div>
                </div>
	        </div>
            )
    }
}

export default RedditPanel