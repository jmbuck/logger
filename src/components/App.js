import React, {Component} from 'react';
import {Redirect, Route, Switch} from 'react-router'
import '../css/common.css';
import Navbar from './Navbar'

import PreferencePanel from "./PreferencePanel"
import FilterPanel from "./FilterPanel"
import DeleteAccountPanel from "./DeleteAccountPanel"
import WebsitePanel from "./WebsitePanel"
import NetflixPanel from "./NetflixPanel"
import RedditPanel from "./RedditPanel"
import DataPanel from './DataPanel'
import YoutubePanel from './YoutubePanel'

class App extends Component {

    constructor(props) {
        super(props)
        this.state = {
            data: {},
            uid: null
        }
    }

    componentWillMount() {
        auth.onAuthStateChanged(
            (user) => {
                (user) => {
                    if(user) {
                        //Finish Signing in
                        this.authHandler(user)
                    } else {
                        //Finish Signing out
                        this.setState({ uid: null })
                    }
                }
            }
        )
    }

    syncData = () => {
        this.ref = base.syncState(
            `data/${this.state.uid}`,
            {
                context: this,
                state: 'data',
            }
        )
    }

    stopSyncing = () => {
        if(this.ref) {
            base.removeBinding(this.ref)
        }
    }

    signedIn = () => {
        return this.state.uid
    }

    signOut = () =>{
        auth
            .signOut()
            .then(() => {
                this.stopSyncing()
                this.setState({ data: {} })
                this.props.history.push('/login')
            })
    }

    authHandler = (user) => {
        this.setState({ uid: user.uid }, this.syncData)
    }

    render() {
        return (
            <div className="App">
            <Navbar {...this.props} signOut={this.signOut} />
            <Switch>
                <Route path='/data' render={(navProps) => <DataPanel {...navProps} />}/>
                <Route path='/preferences' render={(navProps) => <PreferencePanel {...navProps} />}/>
                <Route path='/filters' render={(navProps) => <FilterPanel {...navProps} />}/>
                <Route path='/deleteaccount' render={(navProps) => <DeleteAccountPanel {...navProps} />}/>
                <Route path='/websites' render={(navProps) => <WebsitePanel {...navProps} />}/>
                <Route path='/netflix' render={(navProps) => <NetflixPanel {...navProps} />}/>
                <Route path='/reddit' render={(navProps) => <RedditPanel {...navProps} />}/>
                <Route path='/login' render={(navProps) => <LoginPanel {...navProps} />}/>
                <Route path='/youtube' render={(navProps) => <YoutubePanel {...navProps} />}/>
                <Route path='/' render={(navProps) => !this.signedIn()? <LoginPanel {...navProps} />: <Redirect to='/data' />}/>
            </Switch>
            </div>
        );
    }
}

export default App;
