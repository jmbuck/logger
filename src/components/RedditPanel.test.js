import React from 'react';
import Enzyme from 'enzyme';
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from "./App";
import { RedditPanel } from './RedditPanel'
import firebase from "firebase";

Enzyme.configure({ adapter: new Adapter() });

describe("App", () => {
    beforeAll(() => {
        firebase.auth = jest.fn().mockReturnValue({
            onAuthStateChanged: true
        });
    });

    it("should create component", () => {
        //const component = Enzyme.shallow(<App />);

        console.log(component);
    });
});


describe('Retrieve user reddit data', () => {
    it('renders correctly', () => {
        const wrapper = shallow(
            <RedditPanel
                data= {
                        {
                            "gifs": {
                                "time" : 4634
                            },
                            "news" : {
                                "time" : 9678
                            },
                            "sports" : {
                                "time" : 1555
                            },
                            "todayilearned" : {
                                "time" : 8818
                            }
                        }
                }
            />,
        );
        expect(wrapper).toMatchSnapshot();
    });
});

