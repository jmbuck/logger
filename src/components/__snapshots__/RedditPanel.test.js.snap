// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Retrieve user reddit data renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RedditPanel
    data={
        Object {
            "gifs": Object {
              "time": 4634,
            },
            "news": Object {
              "time": 9678,
            },
            "sports": Object {
              "time": 1555,
            },
            "todayilearned": Object {
              "time": 8818,
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="panel-container"
>
        <div
                className="panel-reddit"
        >
                <DataNav
                        data={
                                Object {
                                        "gifs": Object {
                                          "time": 4634,
                                        },
                                        "news": Object {
                                          "time": 9678,
                                        },
                                        "sports": Object {
                                          "time": 1555,
                                        },
                                        "todayilearned": Object {
                                          "time": 8818,
                                        },
                                      }
                        }
                />
                <div
                        className="panel-center-content"
                >
                        <h1>
                                Reddit
                        </h1>
                        <table
                                border="1px solid black"
                        >
                                <tr>
                                        <th>
                                                Subreddit
                                        </th>
                                        <th>
                                                Time Spent
                                        </th>
                                        <th>
                                                Delete
                                        </th>
                                </tr>
                        </table>
                </div>
        </div>
</div>,
      "className": "panel",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="panel-reddit"
>
          <DataNav
                    data={
                              Object {
                                        "gifs": Object {
                                          "time": 4634,
                                        },
                                        "news": Object {
                                          "time": 9678,
                                        },
                                        "sports": Object {
                                          "time": 1555,
                                        },
                                        "todayilearned": Object {
                                          "time": 8818,
                                        },
                                      }
                    }
          />
          <div
                    className="panel-center-content"
          >
                    <h1>
                              Reddit
                    </h1>
                    <table
                              border="1px solid black"
                    >
                              <tr>
                                        <th>
                                                  Subreddit
                                        </th>
                                        <th>
                                                  Time Spent
                                        </th>
                                        <th>
                                                  Delete
                                        </th>
                              </tr>
                    </table>
          </div>
</div>,
        "className": "panel-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <DataNav
              data={
                            Object {
                                          "gifs": Object {
                                            "time": 4634,
                                          },
                                          "news": Object {
                                            "time": 9678,
                                          },
                                          "sports": Object {
                                            "time": 1555,
                                          },
                                          "todayilearned": Object {
                                            "time": 8818,
                                          },
                                        }
              }
/>,
            <div
              className="panel-center-content"
>
              <h1>
                            Reddit
              </h1>
              <table
                            border="1px solid black"
              >
                            <tr>
                                          <th>
                                                        Subreddit
                                          </th>
                                          <th>
                                                        Time Spent
                                          </th>
                                          <th>
                                                        Delete
                                          </th>
                            </tr>
              </table>
</div>,
          ],
          "className": "panel-reddit",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "data": Object {
                "gifs": Object {
                  "time": 4634,
                },
                "news": Object {
                  "time": 9678,
                },
                "sports": Object {
                  "time": 1555,
                },
                "todayilearned": Object {
                  "time": 8818,
                },
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  Reddit
</h1>,
                <table
                  border="1px solid black"
>
                  <tr>
                                    <th>
                                                      Subreddit
                                    </th>
                                    <th>
                                                      Time Spent
                                    </th>
                                    <th>
                                                      Delete
                                    </th>
                  </tr>
</table>,
              ],
              "className": "panel-center-content",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Reddit",
                },
                "ref": null,
                "rendered": "Reddit",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "border": "1px solid black",
                  "children": Array [
                    <tr>
                      <th>
                                            Subreddit
                      </th>
                      <th>
                                            Time Spent
                      </th>
                      <th>
                                            Delete
                      </th>
</tr>,
                    Array [],
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <th>
                          Subreddit
</th>,
                        <th>
                          Time Spent
</th>,
                        <th>
                          Delete
</th>,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Subreddit",
                        },
                        "ref": null,
                        "rendered": "Subreddit",
                        "type": "th",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Time Spent",
                        },
                        "ref": null,
                        "rendered": "Time Spent",
                        "type": "th",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Delete",
                        },
                        "ref": null,
                        "rendered": "Delete",
                        "type": "th",
                      },
                    ],
                    "type": "tr",
                  },
                ],
                "type": "table",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="panel-container"
>
          <div
                    className="panel-reddit"
          >
                    <DataNav
                              data={
                                        Object {
                                                  "gifs": Object {
                                                    "time": 4634,
                                                  },
                                                  "news": Object {
                                                    "time": 9678,
                                                  },
                                                  "sports": Object {
                                                    "time": 1555,
                                                  },
                                                  "todayilearned": Object {
                                                    "time": 8818,
                                                  },
                                                }
                              }
                    />
                    <div
                              className="panel-center-content"
                    >
                              <h1>
                                        Reddit
                              </h1>
                              <table
                                        border="1px solid black"
                              >
                                        <tr>
                                                  <th>
                                                            Subreddit
                                                  </th>
                                                  <th>
                                                            Time Spent
                                                  </th>
                                                  <th>
                                                            Delete
                                                  </th>
                                        </tr>
                              </table>
                    </div>
          </div>
</div>,
        "className": "panel",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="panel-reddit"
>
            <DataNav
                        data={
                                    Object {
                                                "gifs": Object {
                                                  "time": 4634,
                                                },
                                                "news": Object {
                                                  "time": 9678,
                                                },
                                                "sports": Object {
                                                  "time": 1555,
                                                },
                                                "todayilearned": Object {
                                                  "time": 8818,
                                                },
                                              }
                        }
            />
            <div
                        className="panel-center-content"
            >
                        <h1>
                                    Reddit
                        </h1>
                        <table
                                    border="1px solid black"
                        >
                                    <tr>
                                                <th>
                                                            Subreddit
                                                </th>
                                                <th>
                                                            Time Spent
                                                </th>
                                                <th>
                                                            Delete
                                                </th>
                                    </tr>
                        </table>
            </div>
</div>,
          "className": "panel-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <DataNav
                data={
                                Object {
                                                "gifs": Object {
                                                  "time": 4634,
                                                },
                                                "news": Object {
                                                  "time": 9678,
                                                },
                                                "sports": Object {
                                                  "time": 1555,
                                                },
                                                "todayilearned": Object {
                                                  "time": 8818,
                                                },
                                              }
                }
/>,
              <div
                className="panel-center-content"
>
                <h1>
                                Reddit
                </h1>
                <table
                                border="1px solid black"
                >
                                <tr>
                                                <th>
                                                                Subreddit
                                                </th>
                                                <th>
                                                                Time Spent
                                                </th>
                                                <th>
                                                                Delete
                                                </th>
                                </tr>
                </table>
</div>,
            ],
            "className": "panel-reddit",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "data": Object {
                  "gifs": Object {
                    "time": 4634,
                  },
                  "news": Object {
                    "time": 9678,
                  },
                  "sports": Object {
                    "time": 1555,
                  },
                  "todayilearned": Object {
                    "time": 8818,
                  },
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1>
                    Reddit
</h1>,
                  <table
                    border="1px solid black"
>
                    <tr>
                                        <th>
                                                            Subreddit
                                        </th>
                                        <th>
                                                            Time Spent
                                        </th>
                                        <th>
                                                            Delete
                                        </th>
                    </tr>
</table>,
                ],
                "className": "panel-center-content",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Reddit",
                  },
                  "ref": null,
                  "rendered": "Reddit",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "border": "1px solid black",
                    "children": Array [
                      <tr>
                        <th>
                                                Subreddit
                        </th>
                        <th>
                                                Time Spent
                        </th>
                        <th>
                                                Delete
                        </th>
</tr>,
                      Array [],
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          <th>
                            Subreddit
</th>,
                          <th>
                            Time Spent
</th>,
                          <th>
                            Delete
</th>,
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "Subreddit",
                          },
                          "ref": null,
                          "rendered": "Subreddit",
                          "type": "th",
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "Time Spent",
                          },
                          "ref": null,
                          "rendered": "Time Spent",
                          "type": "th",
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "Delete",
                          },
                          "ref": null,
                          "rendered": "Delete",
                          "type": "th",
                        },
                      ],
                      "type": "tr",
                    },
                  ],
                  "type": "table",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
